### PR TITLE
Additional control flow tests for `for..of` statements

### PR DESCRIPTION
**Pull request by jugglinmike**
V8's MJsUnit test suite did not cover all the relevant cases relating to control flow in the execution of a `for..of` statement's body. These tests increase that coverage.

I've also improved taken this opportunity to make some general improvements to the existing tests:

- organize under a dedicated "control flow" sub-directory
- more granular assertion messages
- more granular assertions for code path execution
- more appropriate variable names

I'm proposing these changes in separate commits to better-facilitate review.

While "on the clock" for the Bocoup-Google contract work, @rwaldron and I will be contributing new tests "&copy; the V8 project authors" as per discussion with @arv.

### CODE REVIEW

File: test/language/for-of/control-flow/break-label-from-catch.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/*---
+es6id: 13.6.4.13 S5.n
+description: >
+    Control flow during body evaluation should honor `break` statements within
+    `try` blocks.
+---*/
+
+function* values() {
+  yield 1;
+  yield 1;
```
Comment (arv) [inline test/language/for-of/control-flow/break-label-from-catch.js]: Maybe add an $ERROR here too?

File: test/language/for-of/control-flow/yield-star-from-catch.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 13.6.4.13
+description: >
+    Control flow during body evaluation should honor `yield` statements within
```
Comment (arv) [inline test/language/for-of/control-flow/yield-star-from-catch.js]: `yield` -> `yield *`

**Comment by arv** [comment]
LGTM

The yield tests could use a second pair of eyes though.

**Comment by rwaldron** [comment]
@arv I also reviewed these a few times over, prior to PR submission.

File: test/language/for-of/control-flow/yield-star-from-catch.js (hunk 2 above)
Comment (jugglinmike) [inline test/language/for-of/control-flow/yield-star-from-catch.js]: Sure thing

File: test/language/for-of/control-flow/break-label-from-catch.js (hunk 1 above)
Comment (jugglinmike) [inline test/language/for-of/control-flow/break-label-from-catch.js]: Sounds good to me. Additionally, the second `yield` statement is extraneous. This was a pattern in a number of tests, so I've fixed it in all cases.

**Comment by jugglinmike** [comment]
Thanks @arv -- I've incorporated all of your feedback.

**Comment by bterlson** [comment]
Let's put this under language/statements/for-of. Also not a fan of the control-flow subdirectory.

**Comment by bterlson** [comment]
Looks like you can add the generators feature to the frontmatter for all these files. Otherwise, good to go I think!

**Comment by jugglinmike** [comment]
@bterlson Alrighty!

- moved all the files under `language/for-of/` to `language/statements/for-of/` (including the previously-existing files).
- moved the contents of the `control-flow` directory to the new `language/statements/for-of/` directory
- added `features: [generators]` to each file's frontmatter

**Comment by bterlson** [comment]
Looks good for squash!

**Comment by jugglinmike** [comment]
@bterlson In this case, I tried to organize the work into logically-distinct commits. If you agree, you can merge from [the original commit history from a separate branch on Bocoup's fork](https://github.com/bocoup/test262/tree/for-of-more-control-flow-pre-squash). Otherwise, you can merge the single commit on this branch (I've force-pushed a single commit that comprises all the code changes and commit messages)

**Comment by bterlson** [comment]
I am OK with many logically distinct commits so long as they're clean :) but for now I will just pull this in!

**Pull request merged**